### PR TITLE
OSSM-3666: migrate maistra test-infra containers jobs

### PR DESCRIPTION
**Pull request by yxun**

### CODE REVIEW

**Comment by yxun** [comment]
It looks like the `make update` target willl automatically remove "postsubmit: true"  in the push-containers-x-x test job section.

File: ci-operator/config/maistra/test-infra/maistra-test-infra-main.yaml [hunk 1]
```diff
@@ -82,6 +82,379 @@ tests:
...
+      env:
+      - name: MAISTRA_NAMESPACE
+      - name: MAISTRA_SC_POD
+      from: src
+      resources:
+        requests:
+          cpu: "2"
+          memory: 2Gi
+      timeout: 1h0m0s
+    - as: push-containers-2-1
```
Comment (jwendell) [inline ci-operator/config/maistra/test-infra/maistra-test-infra-main.yaml:143]: push jobs need to be on their own level. They will be postsubmit jobs

**Review by jwendell — CHANGES_REQUESTED** [review]

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-test-infra-main-test-infra-push-containers-2-3

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-test-infra-main-test-infra-build-containers-2-3

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-test-infra-main-test-infra-push-containers-2-3

**Review by jwendell — COMMENTED** [review]
LGTM, put the `postsubmit: true` back

**Review by jwendell — CHANGES_REQUESTED** [review]
Please use `run_if_changed` as we have today in test-infra jobs. Sorry for not having noticed it before.

**Review by jwendell — APPROVED** [review]

**Pull request merged**